src/Microsoft.AspNet.Identity/RoleManager.cs

```diff
@@ -404,12 +404,12 @@ private IRoleClaimStore<TRole> GetClaimStore()
         /// <param name="user"></param>
         /// <param name="methodName"></param>
         /// <returns></returns>
-        protected async Task<IdentityResult> LogResultAsync(IdentityResult result,
+        protected Task<IdentityResult> LogResultAsync(IdentityResult result,
             TRole role, [System.Runtime.CompilerServices.CallerMemberName] string methodName = "")
         {
-            result.Log(Logger, Resources.FormatLoggingResultMessageForRole(methodName, await GetRoleIdAsync(role)));
+            result.Log(Logger, () => { return Resources.FormatLoggingResultMessageForRole(methodName, GetRoleIdAsync(role).Result); });
```

Comment:
Yo are missing my point. Try `() => Resources.FormatLoggingResultMessageForRole(...)`. Single line lambdas don't need `return` to return something.